debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/pointer/helper/ValueTypeHelper.java

```diff
@@ -0,0 +1,421 @@
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+package com.ibm.j9ddr.vm29.pointer.helper;
+
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
+import java.lang.invoke.MethodHandles.Lookup;
+import java.lang.invoke.MethodType;
+import java.lang.invoke.WrongMethodTypeException;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import com.ibm.j9ddr.CorruptDataException;
+import com.ibm.j9ddr.logging.LoggerNames;
+import com.ibm.j9ddr.vm29.j9.J9ConstantHelper;
+import com.ibm.j9ddr.vm29.pointer.AbstractPointer;
+import com.ibm.j9ddr.vm29.pointer.StructurePointer;
+import com.ibm.j9ddr.vm29.pointer.generated.J9BuildFlags;
+import com.ibm.j9ddr.vm29.pointer.generated.J9ClassPointer;
+import com.ibm.j9ddr.vm29.pointer.generated.J9ROMClassPointer;
+import com.ibm.j9ddr.vm29.pointer.generated.J9ROMNameAndSignaturePointer;
+import com.ibm.j9ddr.vm29.structure.J9JavaAccessFlags;
+import com.ibm.j9ddr.vm29.structure.J9JavaClassFlags;
+import com.ibm.j9ddr.vm29.types.UDATA;
+
+/**
+ * Value type helper class
+ *
+ * areValueTypesSupported() must return true before using any other methods in this class
+ */
+public class ValueTypeHelper {
+	protected static final Logger logger = Logger.getLogger(LoggerNames.LOGGER_INTERACTIVE_CONTEXT);
+
+	private static ValueTypeHelper helper = null;
+
+	private static class ValueTypeSupportEnabledHelper extends ValueTypeHelper {
+		private static final long J9AccValueType = J9ConstantHelper.getLong(J9JavaAccessFlags.class, "J9AccValueType", 0);
+		private static final long J9ClassIsValueType = J9ConstantHelper.getLong(J9JavaClassFlags.class, "J9ClassIsValueType", 0);
+		private static final long J9ClassLargestAlignmentConstraintDouble = J9ConstantHelper.getLong(J9JavaClassFlags.class, "J9ClassLargestAlignmentConstraintDouble", 0);
+		private static final long J9ClassLargestAlignmentConstraintReference = J9ConstantHelper.getLong(J9JavaClassFlags.class, "J9ClassLargestAlignmentConstraintReference", 0);
+		private static final long J9ClassIsFlattened = J9ConstantHelper.getLong(J9JavaClassFlags.class, "J9ClassIsFlattened", 0);
+		private MethodHandle getFlattenedClassCachePointer = null;
+		private Class<?> flattenedClassCachePointer = null;
+		private MethodHandle flattenedClassCache_numberOfEntries = null;
+		private MethodHandle flattenedClassCacheEntry_NaS = null;
+		private MethodHandle flattenedClassCacheEntry_clazz = null;
+		private MethodHandle flattenedClassCacheEntry_cast = null;
+		private Class<?> flattenedClassCacheEntryPointer = null;
+
+		ValueTypeSupportEnabledHelper() {
+			super();
+			try {
+				Lookup lookup = MethodHandles.lookup();
+				flattenedClassCachePointer = Class.forName("com.ibm.j9ddr.vm29.pointer.generated.J9FlattenedClassCachePointer");
```

Comment:
Note: this fails in builds without support for value types; that problem will disappear once we start generating pointer classes at runtime, derived from the blob in the core file.